### DISCUSSION TITLE
Add Metadata to ArrowFiles

### DISCUSSION THREAD

**Discussion by Desi0** [GreenleafLab/ArchR#788] [Questions / Documentation]
I am wondering how to add metadata to my ArchR Arrow files.
I have samples from different groups (e.g. different ages, treatment + control etc) but I cannot manage to add these kind of information. I think the easiest way to add such metadata is to do so when creating the Arrow files. But how?
Can somebody help me?
Thanks.

**Comment by rcorces** [comment]
Metadata is typically added to the `ArchRProject` via the `cellColData` slot.
https://www.archrproject.com/reference/addCellColData.html

see "Adding data to an ArchRProject" - https://www.archrproject.com/bookdown/manipulating-an-archrproject.html

**Reply by Desi0** [reply]
Thanks for the fast reply. 
I tried both methods before, the _$_ accessor and _addCellColData_

The $ accessor did not work out for me. My metadata was overwritten for all cells instead of just for the samples I wished to change.

For addCellColData, I need a list of cell names from all control samples and one for all treated samples, right? And then run this command two times?

That will get quite laborious if you have more then 2 groups and different metadata to add. Isn't there an easier/faster way to add metadata? 
Just as a suggestion: Wouldn't it be possible to read in a csv together with the fragments when creating the arrow files?

**Reply by rcorces** [reply]
> The $ accessor did not work out for me. My metadata was overwritten for all cells instead of just for the samples I wished to change.

This is hard for me to interpret what you mean but perhaps you can show how you are trying to add metadata

> Isn't there an easier/faster way to add metadata? Wouldn't it be possible to read in a csv together with the fragments when creating the arrow files?

A simple for loop should work. The problem with supporting requests like this is that it requires the metadata to be in a very precise format and I think we would rather leave this up to the end user.

```
#make a dummy dataframe with sample names as rownames to be used for loading metadata into project
projHeme5 <- loadArchRProject(path = "./Save-ProjHeme5/")
meta_df <- data.frame(varA = c(1,2,3), varB = c("X","Y","Z"), stringsAsFactors = FALSE)
rownames(meta_df) <- unique(projHeme5@cellColData$Sample)

for (y in 1:ncol(meta_df)){
  projHeme5@cellColData[,colnames(meta_df)[y]] <- NA
  for(x in 1:nrow(meta_df)) {
    cellsToChange <- getCellNames(ArchRProj = projHeme5)[which(as.character(projHeme5@cellColData$Sample) == rownames(meta_df)[x])]
    projHeme5@cellColData[cellsToChange,colnames(meta_df)[y]] <- meta_df[x,y]
  }
}
```
